sdk/core/azure-core/inc/http/bodyStream.hpp

```diff
@@ -0,0 +1,269 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#pragma once
+
+#ifdef Posix
+#include <unistd.h>
+#endif
+
+#ifdef Windows
+#define WIN32_LEAN_AND_MEAN
+#define NOMINMAX
+#include <Windows.h>
+#endif // Windows
+
+#include <algorithm>
+#include <context.hpp>
+#include <cstdint>
+#include <cstdio>
+#include <cstring>
+#include <memory>
+#include <vector>
+
+namespace Azure { namespace Core { namespace Http {
+
+  // BodyStream is used to read data to/from a service
+  class BodyStream {
+  public:
+    virtual ~BodyStream() = 0;
+
+    // Returns the length of the data; used with the HTTP Content-Length header
+    virtual int64_t Length() const = 0;
+
+    // Resets the stream back to the beginning (for retries)
+    // Derived classes that send data in an HTTP request MUST override this and implement it
+    // properly.
+    virtual void Rewind()
+    {
+      throw "Not Implemented"; // TODO: Replace with best practice as defined by guideline
+    };
+
+    // Reads more data; throws if error/canceled
+    // return copied size
+    virtual int64_t Read(Context& context, uint8_t* buffer, int64_t count) = 0;
+
+    // Closes the stream; typically called after all data read or if an error occurs.
+    virtual void Close() = 0;
+
+    // Keep reading until buffer is all fill out of the end of stream content is reached
+    static int64_t ReadToCount(Context& context, BodyStream& body, uint8_t* buffer, int64_t count)
+    {
+      int64_t readBytes;
```

Comment:
`readBytes` isn't used out of the for loop, so maybe you want to move it into the loop